client/mm/libxc/interface.go

```diff
@@ -0,0 +1,96 @@
+package libxc
+
+import (
+	"context"
+	"fmt"
+
+	"decred.org/dcrdex/dex"
+)
+
+// ExchangeBalance holds the available and locked balances of an asset
+// on a CEX.
+type ExchangeBalance struct {
+	Available uint64 `json:"available"`
+	Locked    uint64 `json:"locked"`
+}
+
+// TradeUpdate is a notification sent when the status of a trade on the CEX
+// has been updated.
+type TradeUpdate struct {
+	TradeID  string
+	Complete bool // cancelled or filled
+}
+
+// Market is the base and quote assets of a market on a CEX.
+type Market struct {
+	BaseID  uint32 `json:"base"`
+	QuoteID uint32 `json:"quote"`
+}
+
+// CEX implements a set of functions that can be used to interact with a
+// centralized exchange's spot trading API. All rates and quantities
+// when interacting with the CEX interface will adhere to the standard
+// rates and quantities of the DEX.
+type CEX interface {
+	dex.Connector
+	// Balance returns the balance of an asset at the CEX.
+	Balance(symbol string) (*ExchangeBalance, error)
+	// Balances returns a list of all asset balances at the CEX. Only assets that are
+	// registered in the DEX client will be returned.
+	Balances() (map[uint32]*ExchangeBalance, error)
+	// CancelTrade cancels a trade on the CEX.
+	CancelTrade(ctx context.Context, baseSymbol, quoteSymbol, tradeID string) error
+	// GenerateTradeID returns a trade ID that must be passed as an argument
+	// when calling Trade. This ID will be used to identify updates to the
+	// trade. It is necessary to pre-generate this because updates to the
+	// trade may be sent before Trade returns.
+	GenerateTradeID() string
```

Comment:
The issue is that the `executionReport` for this Trade may arrive before the response to the `Trade` request arrives. In this case, the caller of `Trade` will not know which trade the notification is about. I think the only choices are to generate the trade id beforehand, or do some more complicated buffering of notifications.

Reply:
But in the only place this is used, nothing is being done with the `tradeID` before passing it to `Trade`. 

https://github.com/decred/dcrdex/blob/cdf15b2417e92920ab7bb468559fbfa823a7e83c/client/mm/mm_simple_arb.go#L272-L273

The `arbSequence` isn't being added to the `activeArbs`, and it doesn't matter because the `activeArbsMtx` handles synchronization, since it's `Lock`ed both here and in `handleCEXTradeUpdate`.